### DISCUSSION TITLE
Any plan to support asciidoc?

### DISCUSSION THREAD

**Discussion by kbyyd24** [HEIGE-PCloud/DoIt#1296] [Ideas]
Some of my posts are asciidoc. I tried to use this theme, there are two issues I found so far:
- TOC not created automatically, have to add `:toc:` in the content to generate it
  - Once created, the TOC is fixed at the top of the page, not stuck on the window.
- Code block highlight doesn't work

TOC before scroll down:
![image](https://github.com/HEIGE-PCloud/DoIt/assets/9191275/615b4186-dfbe-4bc8-b7da-579559feeb0a)

TOC after scroll down:
![image](https://github.com/HEIGE-PCloud/DoIt/assets/9191275/b11b40d5-a6ff-4600-9eb1-086e3fde5a30)

Code block:
![image](https://github.com/HEIGE-PCloud/DoIt/assets/9191275/05009dfe-87a6-43b3-9ff5-558e75b34dd2)

**Comment by HEIGE-PCloud** [comment]
DoIt heavily relies on [Hugo render hooks](https://gohugo.io/render-hooks/introduction/) to provide some of its core functionalities (images, headers, code blocks). Unfortunately, such render hooks do not work with asciidoc. So none of the layouts here will be executed: https://github.com/HEIGE-PCloud/DoIt/tree/90c35b5840e924988a160be39a7de6982e8445a5/layouts/_default/_markup

This is a limitation of Hugo, there isn't much we can do about it as a theme. 

You might workaround some of the issues by configuring asciidoctor https://gohugo.io/getting-started/configuration-markup/#asciidoc

https://github.com/gohugoio/hugo/issues?q=is%3Aissue+is%3Aopen+asciidoc
